flytekit/types/file/file.py

```diff
@@ -372,6 +372,9 @@ def __repr__(self):
     def __str__(self):
         return self.path
 
+    def __hash__(self):
+        return hash(self._serialize()["path"])
```

Comment:
Can we add a small test where we place `FlyteFile` into a set or a dict? (To emulate what Jupyter does without having to have add a new full integration test)